citro3d-sys/src/bindings.rs

```diff
@@ -80,948 +82,219 @@ where
-}
-extern "C" {
-    #[must_use]
-    #[doc = " @brief Gets a bool uniform of a shader."]
-    #[doc = " @param si Shader instance to use."]
-    #[doc = " @param id ID of the bool uniform."]
-    #[doc = " @param value Pointer to output the value to."]
-    pub fn shaderInstanceGetBool(
-        si: *mut shaderInstance_s,
-        id: ::libc::c_int,
-        value: *mut bool,
-    ) -> Result;
-}
-extern "C" {
-    #[doc = " @brief Gets the location of a shader's uniform."]
-    #[doc = " @param si Shader instance to use."]
-    #[doc = " @param name Name of the uniform."]
-    pub fn shaderInstanceGetUniformLocation(
-        si: *mut shaderInstance_s,
-        name: *const ::libc::c_char,
-    ) -> s8;
-}
-extern "C" {
-    #[must_use]
-    #[doc = " @brief Initializes a shader program."]
-    #[doc = " @param sp Shader program to initialize."]
-    pub fn shaderProgramInit(sp: *mut shaderProgram_s) -> Result;
-}
-extern "C" {
-    #[must_use]
-    #[doc = " @brief Frees a shader program."]
-    #[doc = " @param sp Shader program to free."]
-    pub fn shaderProgramFree(sp: *mut shaderProgram_s) -> Result;
-}
-extern "C" {
-    #[must_use]
-    #[doc = " @brief Sets the vertex shader of a shader program."]
-    #[doc = " @param sp Shader program to use."]
-    #[doc = " @param dvle Vertex shader to set."]
-    pub fn shaderProgramSetVsh(sp: *mut shaderProgram_s, dvle: *mut DVLE_s) -> Result;
-}
-extern "C" {
-    #[must_use]
-    #[doc = " @brief Sets the geometry shader of a shader program."]
-    #[doc = " @param sp Shader program to use."]
-    #[doc = " @param dvle Geometry shader to set."]
-    #[doc = " @param stride Input stride of the shader (pass 0 to match the number of outputs of the vertex shader)."]
-    pub fn shaderProgramSetGsh(sp: *mut shaderProgram_s, dvle: *mut DVLE_s, stride: u8_) -> Result;
-}
-extern "C" {
-    #[must_use]
-    #[doc = " @brief Configures the permutation of the input attributes of the geometry shader of a shader program."]
-    #[doc = " @param sp Shader program to use."]
-    #[doc = " @param permutation Attribute permutation to use."]
-    pub fn shaderProgramSetGshInputPermutation(
-        sp: *mut shaderProgram_s,
-        permutation: u64_,
-    ) -> Result;
+pub struct _rand48 {
+    pub _seed: [::libc::c_ushort; 3usize],
+    pub _mult: [::libc::c_ushort; 3usize],
+    pub _add: ::libc::c_ushort,
 }
-extern "C" {
-    #[must_use]
-    #[doc = " @brief Configures the shader units to use the specified shader program."]
-    #[doc = " @param sp Shader program to use."]
-    #[doc = " @param sendVshCode When true, the vertex shader's code and operand descriptors are uploaded."]
-    #[doc = " @param sendGshCode When true, the geometry shader's code and operand descriptors are uploaded."]
-    pub fn shaderProgramConfigure(
-        sp: *mut shaderProgram_s,
-        sendVshCode: bool,
-        sendGshCode: bool,
-    ) -> Result;
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct _reent {
```

Comment:
Hmm, looks like the `allowlist_file` might still need a little tweaking since this stuff is here...